core/services/ocr2/plugins/directrequestocr/reporting.go

```diff
@@ -0,0 +1,225 @@
+}
+
+// Query() complies with ReportingPlugin
+func (r *directRequestReporting) Query(ctx context.Context, ts types.ReportTimestamp) (types.Query, error) {
+	r.logger.Debug("directRequestReporting Query phase", commontypes.LogFields{
+		"epoch": ts.Epoch,
+		"round": ts.Round,
+	})
+	// TODO add batch size to config
+	// https://app.shortcut.com/chainlinklabs/story/56615/config-for-reporting-plugin
+	const maxBatchSize = 10
+	results, err := r.pluginORM.FindOldestEntriesByState(directrequestocr.RESULT_READY, maxBatchSize)
+	if err != nil {
+		return nil, err
+	}
+
+	queryProto := Query{}
+	for _, result := range results {
+		queryProto.RequestIDs = append(queryProto.RequestIDs, result.ContractRequestID[:])
+	}
+	return proto.Marshal(&queryProto)
+}
+
+// Observation() complies with ReportingPlugin
+func (r *directRequestReporting) Observation(ctx context.Context, ts types.ReportTimestamp, query types.Query) (types.Observation, error) {
+	r.logger.Debug("directRequestReporting Observation phase", commontypes.LogFields{
+		"epoch": ts.Epoch,
+		"round": ts.Round,
+	})
+
+	queryProto := &Query{}
+	err := proto.Unmarshal(query, queryProto)
+	if err != nil {
+		return nil, err
+	}
+
+	observationProto := Observation{}
+	for _, id := range queryProto.RequestIDs {
+		localResult, _ := r.pluginORM.FindById(sliceToByte32(id))
+		if localResult.State == directrequestocr.RESULT_READY {
+			resultProto := ProcessedRequest{
+				RequestID: localResult.ContractRequestID[:],
+				Result:    localResult.Result,
+				Error:     []byte(localResult.Error),
+			}
+			observationProto.ProcessedRequests = append(observationProto.ProcessedRequests, &resultProto)
+		}
+	}
+	r.logger.Debug("directRequestReporting Observation phase done", commontypes.LogFields{
+		"nReadyRequests": len(observationProto.ProcessedRequests),
+	})
+
+	return proto.Marshal(&observationProto)
+}
+
+// Report() complies with ReportingPlugin
+func (r *directRequestReporting) Report(ctx context.Context, ts types.ReportTimestamp, query types.Query, obs []types.AttributedObservation) (bool, types.Report, error) {
+	r.logger.Debug("directRequestReporting Report phase", commontypes.LogFields{
+		"epoch":         ts.Epoch,
+		"round":         ts.Round,
+		"nObservations": len(obs),
+	})
+
+	queryProto := &Query{}
+	err := proto.Unmarshal(query, queryProto)
+	if err != nil {
+		r.logger.Error("directRequestReporting Report phase unable to decode query!",
+			commontypes.LogFields{"err": err})
+		return false, nil, err
+	}
+
+	reqIdToResultList := make(map[string][]*ProcessedRequest)
+	for _, id := range queryProto.RequestIDs {
+		reqIdToResultList[string(id)] = []*ProcessedRequest{}
+	}
+
+	for _, ob := range obs {
+		observationProto := &Observation{}
+		err = proto.Unmarshal(ob.Observation, observationProto)
+		if err != nil {
+			r.logger.Error("directRequestReporting Report phase unable to decode observation!",
+				commontypes.LogFields{"err": err, "observer": ob.Observer})
+			continue
```

Comment:
In this case, we're not even able to decode which request this observations belongs to. So I don't see how we can send it forward... The error here would most likely indicate a deeper system issue.

Reply:
I was thinking more towards the direction that this indicates that the replica which sent an observation that cannot be decoded might be considered faulty.
But there is not much we can do this information and since there is a threshold (now 3) for observations per request, I guess it's fine ignoring the problematic observation.